service/src/test/java/io/camunda/service/query/filter/AuthorizationFilterTest.java

```diff
@@ -0,0 +1,107 @@
+
+import io.camunda.search.clients.query.SearchTermQuery;
+import io.camunda.service.AuthorizationServices;
+import io.camunda.service.entities.AuthorizationEntity;
+import io.camunda.service.search.filter.AuthorizationFilter;
+import io.camunda.service.search.filter.AuthorizationFilter.Builder;
+import io.camunda.service.search.filter.FilterBuilders;
+import io.camunda.service.search.query.SearchQueryBuilders;
+import io.camunda.service.util.StubbedBrokerClient;
+import io.camunda.service.util.StubbedCamundaSearchClient;
+import io.camunda.util.ObjectBuilder;
+import io.camunda.zeebe.protocol.impl.record.value.authorization.AuthorizationRecord;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class AuthorizationFilterTest {
+  private AuthorizationServices<AuthorizationRecord> services;
+  private StubbedCamundaSearchClient client;
+  private StubbedBrokerClient brokerClient;
+
+  @BeforeEach
+  public void before() {
+    client = new StubbedCamundaSearchClient();
+    new AuthorizationSearchQueryStub().registerWith(client);
+    services = new AuthorizationServices<>(brokerClient, client);
+  }
+
+  @Test
+  public void emptyQueryReturnsAllResults() {
+    // given
+    final AuthorizationFilter filter = new AuthorizationFilter.Builder().build();
+    final var searchQuery = SearchQueryBuilders.authorizationSearchQuery((b) -> b.filter(filter));
+
+    // when
+    final var searchQueryResult = services.search(searchQuery);
+
+    // then
+    assertThat(searchQueryResult.total()).isEqualTo(2);
+    assertThat(searchQueryResult.items()).hasSize(2);
+    final AuthorizationEntity item = searchQueryResult.items().get(0);
+    assertThat(item.value().ownerKey()).isEqualTo("username1");
+  }
+
+  @ParameterizedTest
+  @MethodSource("queryFilterParameters")
+  public void shouldQueryByField(
+      final Function<Builder, ObjectBuilder<AuthorizationFilter>> fn,
+      final String column,
+      final String value) {
+    // given
+    final var authorizationFilter = FilterBuilders.authorization(fn);
+    final var searchQuery =
+        SearchQueryBuilders.authorizationSearchQuery(q -> q.filter(authorizationFilter));
+
+    // when
+    services.search(searchQuery);
+
+    // then
+    final var searchRequest = client.getSingleSearchRequest();
+
+    final var queryVariant = searchRequest.query().queryOption();
+    assertThat(queryVariant)
+        .isInstanceOfSatisfying(
+            SearchTermQuery.class,
+            t -> {
+              assertThat(t.field()).isEqualTo(column);
+              assertThat(t.value().stringValue()).isEqualTo(value);
+            });
+  }
+
+  public static Stream<Arguments> queryFilterParameters() {
+    return Stream.of(
+        Arguments.of(
```

Comment:
❓In my idea, it's ok to search by resourceType or ownerType or resourceKey+resourceType and ownerType+ownerKey and combination of them, but it's better to not allow to search by ownerKey without ownerType and also same for resourceKey. wdyt?

Reply:
Good question, so keys in Zeebe are unique so there wouldn't be a case where the same key is used for both an owner or a resource so its fine here to allow searching by one without a contextual type filter :)